qa/tasks/cephfs/test_volumes.py

```diff
@@ -8165,6 +8165,57 @@ def test_unparseable_metafile_on_legacy_to_v1_upgrade(self):
         # remove group
         self._fs_cmd("subvolumegroup", "rm", self.volname, group)
 
+    def test_dangling_symlink(self):
+        """
+        test_dangling_symlink
+        Tests for the presence of any dangling symlink, if yes,
+        will remove it.
+        """
+        subvolume = self._gen_subvol_name()
+        snapshot = self._gen_subvol_snap_name()
+        clone = self._gen_subvol_clone_name()
+
+        # create subvolume in group
+        self._fs_cmd("subvolume", "create", self.volname, subvolume)
+
+        # create subvolume snapshot
+        self._fs_cmd("subvolume", "snapshot", "create", self.volname, subvolume, snapshot)
+
+        # insert delay at the beginning of snapshot clone
+        self.config_set('mgr', 'mgr/volumes/snapshot_clone_delay', 60)
+
+        # disable "capped" clones
+        self.config_set('mgr', 'mgr/volumes/snapshot_clone_no_wait', False)
+
+        # clone
+        self._fs_cmd("subvolume", "snapshot", "clone", self.volname, subvolume, snapshot, clone)
+
+        # list snapshot info
+        result = json.loads(self._fs_cmd("subvolume", "snapshot", "info", self.volname, subvolume, snapshot))
+
+        # verify snapshot info
+        self.assertEqual(result['has_pending_clones'], "yes")
+
+        # remove clone
+        path = os.path.join(".", "volumes", "_nogroup", clone)
```

Comment:
s/path/clone_path for sake of better readability.
 
 And calling os.path.join() is not really needed since we most path components literal strings.
 
```suggestion
        clone_path = f'./volumes/_nogroup/{clone}'
```